code/drasil-printers/lib/Language/Drasil/Markdown/Print.hs

```diff
@@ -0,0 +1,344 @@
+    mathEqn = mjDelimDisp $ rndr contents
+    rndr (E e) = pExpr e
+    rndr c = pSpec rm c
+printLO rm (Table _ rows r b t)  = makeTable rm rows (pSpec rm r) b (pSpec rm t)
+printLO rm (Definition _ ssPs l) = makeDefn rm ssPs (pSpec rm l) <> nl
+printLO rm (List t)              = makeList rm t 0 <> nl
+printLO rm (Figure r c f _)      = makeFigure (pSpec rm r) (pSpec rm c) (text f)
+printLO rm (Bib bib)             = makeBib rm bib
+printLO _ Graph {}               = empty 
+printLO _ CodeBlock {}           = empty
+
+-- | Helper for rendering LayoutObj into Markdown Tables
+printLO' :: RefMap -> LayoutObj -> Doc
+printLO' rm e = stripStr (printLO rm e) nl
+
+-----------------------------------------------------------------
+----------------------- SPEC PRINTING ---------------------------
+-----------------------------------------------------------------
+
+-- | Helper for rendering Specs into Markdown
+pSpec :: RefMap -> Spec -> Doc
+pSpec _ (E e)      = text "\\\\(" <> pExpr e <> text "\\\\)" -- symbols used
+pSpec rm (a :+: b) = pSpec rm a <> pSpec rm b
+pSpec _ HARDNL     = nl
+pSpec rm (Ref Internal       r a) = reflink     rm r (pSpec rm a)
+pSpec rm (Ref (Cite2 EmptyS) r a) = reflink     rm r (pSpec rm a)
+pSpec rm (Ref (Cite2 n)      r a) = reflinkInfo rm r (pSpec rm a) (pSpec rm n)
+pSpec rm (Ref External       r a) = reflinkURI  (text r) (pSpec rm a)
+pSpec rm (Quote q) = doubleQuotes $ pSpec rm q
+pSpec _ s          = HTML.pSpec s 
+
+-----------------------------------------------------------------
+-------------------- EXPRESSION PRINTING ------------------------
+-----------------------------------------------------------------
+
+-- | Helper for rendering Exprs into mdBook compatiable Mathjax
+pExpr :: Expr -> Doc
+pExpr (Str s)        = printMath $ toText $ pure $ lq <> text s <> rq
+  where 
+    lq = text "\\\\(\\``\\\\)"
+    rq = text "''"
+pExpr (Div n d)      = printMath $ command2D "frac" (pExpr' n) (pExpr' d)
+pExpr (Case ps)      = printMath $ mkEnv "cases" (TeX.cases ps lnl pExpr')
+pExpr (Mtx a)        = stripStr (printMath $ mkEnv "bmatrix" (TeX.pMatrix a lnl pExpr')) tab
+pExpr (Row [x])      = br $ pExpr x 
+pExpr (Row l)        = foldl1 (<>) (map pExpr l)
+pExpr (Label s)      = printMath $ TeX.pExpr (Label s')
+  where s' = replace "*" "\\*" (replace "_" "\\_" s)
+pExpr (Sub e)        = bslash <> unders <> br (pExpr e)
+pExpr (Sup e)        = hat    <> br (pExpr e)
+pExpr (Over Hat s)   = printMath $ commandD "hat" (pExpr' s)
+pExpr (MO Perc)      = bslash <> printMath (TeX.pExpr (MO Perc))
+pExpr (Fenced l r m) = fence TeX.Open l <> pExpr m <> fence TeX.Close r
+  where 
+    fence _ Abs = text "\\|"
+    fence a b   = printMath $ TeX.fence a b
+pExpr (Font Bold e)  = printMath $ commandD "boldsymbol" (pExpr' e)
+pExpr (Font Emph e)  = pExpr e
+pExpr (Spc Thin)     = text "\\\\,"
+pExpr (Sqrt e)       = printMath $ commandD "sqrt" (pExpr' e)
+pExpr e              = printMath $ TeX.pExpr e
```

Comment:
I had a discussion with @balacij about this, I will summarize it here.

At first glance, the code for `pExpr` seems very similar to the one in [`Language.Drasil.TeX.Print`](https://github.com/JacquesCarette/Drasil/blob/88a109896dbccd86b0a148837adea09cb8b556a7/code/drasil-printers/lib/Language/Drasil/TeX/Print.hs#L101-L124) and one may assume that it is duplicated. However, there is a reason for this. The differences in expressions between pure LaTeX and the version of Mathjax that mdBook uses is very minor. Specifically, the differences are extra escape characters, different command names (i.e. boldsymbol), quotes, etc. I took the approach of re-defining all the expressions that are different, then delegating to the TeX function in a catch all statement. But since the `pExpr` functions involve recursive calls I had to re-define those as well. This is because once `TeX.pExpr` is called, there is no way for it to return to the Markdown specific `pExpr`.

The reason I went with this approach rather than another (i.e. adding a new parameter to `TeX.pExpr`) is because it is simple and the least amount of code. Adding an additional parameter to `pExpr` that takes in the recursive function call would result in more functions to be added, such as one to begin the recursion. Additionally since all of the printers leverage `TeX.pExpr`, this change would propagate to all the printers.

I will annotate the code below with my reasoning to why those patterns are re-defined:

```Haskell
-- | Helper for rendering Exprs into mdBook compatiable Mathjax
pExpr :: Expr -> Doc
pExpr (Str s)        = printMath $ toText $ pure $ lq <> text s <> rq -- | quotes are different
  where 
    lq = text "\\\\(\\``\\\\)"
    rq = text "''"
pExpr (Div n d)      = printMath $ command2D "frac" (pExpr' n) (pExpr' d) -- | recursive call
pExpr (Case ps)      = printMath $ mkEnv "cases" (TeX.cases ps lnl pExpr') -- | different parameters to TeX.cases 
pExpr (Mtx a)        = stripStr (printMath $ mkEnv "bmatrix" (TeX.pMatrix a lnl pExpr')) tab -- | different parameters to TeX.pMatrix
pExpr (Row [x])      = br $ pExpr x -- | recursive call
pExpr (Row l)        = foldl1 (<>) (map pExpr l) -- | recursive call
pExpr (Label s)      = printMath $ TeX.pExpr (Label s')  -- | escaping special chars in string
  where s' = replace "*" "\\*" (replace "_" "\\_" s)
pExpr (Sub e)        = bslash <> unders <> br (pExpr e) -- | extra escape char
pExpr (Sup e)        = hat    <> br (pExpr e) -- | recursive call
pExpr (Over Hat s)   = printMath $ commandD "hat" (pExpr' s) -- | recursive call
pExpr (MO Perc)      = bslash <> printMath (TeX.pExpr (MO Perc)) -- | extra escape char
pExpr (Fenced l r m) = fence TeX.Open l <> pExpr m <> fence TeX.Close r -- | different fence and recursive call
  where 
    fence _ Abs = text "\\|"
    fence a b   = printMath $ TeX.fence a b
pExpr (Font Bold e)  = printMath $ commandD "boldsymbol" (pExpr' e) -- | change to 'boldsymbol'
pExpr (Font Emph e)  = pExpr e -- | recursive call
pExpr (Spc Thin)     = text "\\\\," -- | extra escape chars
pExpr (Sqrt e)       = printMath $ commandD "sqrt" (pExpr' e) -- | recursive call
pExpr e              = printMath $ TeX.pExpr e -- | catch all statement
```

The annotation above is not in the actual code, please let me know if it is beneficial to add it to the source code as well, along with key points from my blurb above.